perceval/rendering/circuit/phys_skin.py

```diff
@@ -167,7 +167,7 @@ def ffconf_shape(self, comp: AFFConfigurator, canvas, mode_style):
 
         # The actual component
         canvas.add_rect((5, 10), 50 * w - 10, 50 * comp.m - 20, fill="lightgreen")
-        canvas.add_text((w * 25, 30 + 50*(comp.m-1)/2), size=10, ta="middle", text=comp.name)
+        canvas.add_text((w * 25, 30 + 50*(comp.m-1)/2), size=10, ta="middle", text=comp.name, max_size=50*w)
```

Comment:
This should be done everywhere we have a box: Unitaries, sub-circuits (maybe others that I'm forgetting)